### ISSUE TITLE
544307D1 - DEAD OR ALIVE 4

### ISSUE THREAD

**Issue by ryu-highabusa** [xenia-canary/game-compatibility#207]
### Validation

- [X] I've read the [FAQ](https://github.com/xenia-canary/xenia-canary/wiki/FAQ).
- [X] The Xenia build used is from a canary branch. (not MLBS/AlexVS/master/pull requests, etc)
- [X] This issue isn't for help (tech support) with Xenia, or a game.
- [X] I've checked if this game doesn't already have a compatibility report.

### Xbox 360 Marketplace link

https://marketplace.xbox.com/en-us/Product/DEAD-OR-ALIVE-4/66acd000-77fe-1000-9115-d802544307d1

### Xenia version

https://github.com/xenia-canary/xenia-canary/commit/6e13258ad48456f0a27a93b5b149ff0f9c572ae2

### Issues

The boss character causes the game to lock-up; happens during the intro sequence and during gameplay. 

Suspect it has to do with her translucent graphical effects; her smoke effect renders correctly. 

Attached is a screenshot of the exact frame the game consistently hangs on during the attract sequence and a log file leading up that moment. 

Doesn't crash, it hangs. 

I left the emulator open for about a minute or so to let it log as much relevant data as possible. 
Log file line# 2,443,894 is where things appear to go wrong. 
The hang happens at the exact moment 0x830b22f8 gets written to. 

Everything is stock except for the following settings in the config:
**clear_memory_page_state = true 
log_level = 3** 

### Log

[xenia_canary_6e13258-doa4bossattractscreencrash.zip](https://github.com/xenia-canary/game-compatibility/files/14641267/xenia_canary_6e13258-doa4bossattractscreencrash.zip)

|	line#	|	log	|
|	----	|	----	|
|	2443894	|	d> F8000008 MmSetAddressProtect(830B22F8, 00001000, 00000404)	|
|	2443895	|	d> 01000010 Make 1CD60000 -> 1F560000 (41943040b) coherent, action = VC	|
|	2443896	|	d> 01000010 Make 1CD60000 -> 1F560000 (41943040b) coherent, action = VC	|
|	2443897	|	d> 01000010 Make 1C78A000 -> 1C7CB000 (266240b) coherent, action = VC | TC	|
|	2443898	|	G> 01000010 Loaded tiled 256x256x1 2D k_8_8_8_8 texture with 1 unpacked mip level, base at 0x1C78A000 (pitch 256, size 0x00040000), mips at 0x00000000 (size 0x00000000)	|
|	2443899	|	d> 01000010 Make 1C78A000 -> 1C7CB000 (266240b) coherent, action = VC | TC	|
|	2443900	|	G> 01000010 Loaded tiled 256x256x1 2D k_8_8_8_8 texture with 1 unpacked mip level, base at 0x1C78A000 (pitch 256, size 0x00040000), mips at 0x00000000 (size 0x00000000)	|
|	2443901	|	d> 01000010 Make 1C78A000 -> 1C7CB000 (266240b) coherent, action = VC | TC	|
|	2443902	|	d> 01000010 Make 1C7CA000 -> 1C80B000 (266240b) coherent, action = VC | TC	|
|	2443903	|	d> F8000008 MmSetAddressProtect(830B22F8, 00001000, 00000402)	|
|	2443904	|	d> F800001C NtResumeThread(F8000024, 7027FE40(FFFFFFFF))	|
|	2443905	|	d> F8000024 NtSuspendThread(F8000024, 702DFE30(FFFFFFFF), ContextArg)	|
|	2443906	|	d> F800001C NtResumeThread(F8000024, 7027FE40(FFFFFFFF))	|
|	2443907	|	d> F8000024 NtSuspendThread(F8000024, 702DFE30(FFFFFFFF), ContextArg)	|
|	2443908	|	d> F800001C NtResumeThread(F8000024, 7027FE40(FFFFFFFF))	|

### Screenshots

![doa4alpha152crash](https://github.com/xenia-canary/game-compatibility/assets/98571310/66384584-9e42-4e4e-afe5-6062c5f03a38)

### Labels

state-hang

**Comment by GhostRequiemX1** [comment]
i'm able to boot the game's start screen with "kyoto in bloom" stage as a backdrop for the most port but if left to run on its own. i'll get the opening "helicopter scene" cinematic run just fine 

but will get graphics issues entering its showcase demonstration, ill still hear the audio un affected. 

but if i eventually press start, save function/selection is broken and main menu is a artifacting corrupted mess  

Xenia Version: https://github.com/xenia-project/xenia/commit/0fcdc12cb9143f671aae1416acd42a55ddaf1fd8

tried on default xenia canary config and then on my tweaked xenia-canary.config its default except for: 

license_mask = 1 
internal_display_resolution = 16  
mount_cache = true

results where both the same.

![1](https://github.com/xenia-canary/game-compatibility/assets/166696292/629a24a0-9e3f-4fa6-ab16-541578d7dd05)
![2](https://github.com/xenia-canary/game-compatibility/assets/166696292/b06cca2a-c0de-4e34-8ce4-44c709b21758)
![3](https://github.com/xenia-canary/game-compatibility/assets/166696292/ca4fef87-65ec-4caa-b90e-e826681a6ea2)
![4](https://github.com/xenia-canary/game-compatibility/assets/166696292/ece8d064-5b7a-4c3c-913b-95140d888aaa)
![5](https://github.com/xenia-canary/game-compatibility/assets/166696292/f31698b0-d0da-4cfd-8f45-fd9e42ec8b02)
![6](https://github.com/xenia-canary/game-compatibility/assets/166696292/44a5b117-b7cf-4825-8fa0-fe3dc39d6a48)

Labels: state-intro, gpu-drawing-corrupt, gpu shading errors

**Comment by ryu-highabusa** [comment]
> i'm able to boot the game's start screen with "kyoto in bloom" stage as a backdrop for the most port but if left to run on its own. i'll get the opening "helicopter scene" cinematic run just fine 



Set "clear_memory_page_state = true" in the config file to fix most graphical glitches in Team Ninja games.

**Comment by GhostRequiemX1** [comment]
### Update

proceeded with changing to "clear_memory_page_state = false" to "clear_memory_page_state" = true

cleared the log, continued to the start screen, got opening cinematic per just last session let it run i got a sec of everything looking normal on Kokoro before going black screen. then i pressed start navigating to "doa 4 story" and selected kasumi playthrough but other wise gameplay and everything looked normal up to my fight with zack, leifang, tina.

### Log

[xenia.log](https://github.com/xenia-canary/game-compatibility/files/14978095/xenia.log)

### Screenshots

![1](https://github.com/xenia-canary/game-compatibility/assets/166696292/ffaa8cbc-2ec5-4e0f-80b3-46191057aa35)
![2](https://github.com/xenia-canary/game-compatibility/assets/166696292/3c7c3226-4f43-4168-840f-5442f9af2cb9)
![3](https://github.com/xenia-canary/game-compatibility/assets/166696292/bef324e6-4c55-4fec-b54d-087bf7b21fb2)
![4](https://github.com/xenia-canary/game-compatibility/assets/166696292/0cccd963-2f35-4756-ac2d-2f56c866178d)
![5](https://github.com/xenia-canary/game-compatibility/assets/166696292/afdd3482-f6a8-4afa-b6c8-0ed105c0de4b)

**Comment by cerroverb** [comment]
i cant get the game to go past main menu myself, neither this or NG2

**Comment by ryu-highabusa** [comment]
xenia-canary build 0ad1e3d58 addresses the boss crashing issue and it is now possible to complete the story mode. 

![image](https://github.com/user-attachments/assets/62ed8e2c-9697-4387-aa3a-8da152d98c06)
![image](https://github.com/user-attachments/assets/e492db6d-8c1d-48d2-9eca-0724ca01597c)

**Comment by cerroverb** [comment]
i can play this and ninja gaiden ii with no issues now just not on linux

**Comment by legolas119** [comment]
Hi, i'm using the latest Canary version ( a6e3d77) and i have this issue:
the game works fine but the soundtrack very often is missing. It is always missing in the opening screen (where it is necessary to push START), into the character select screen and into the game mode section.
During the game, in some stages, sometimes, the soundtrack is missing: sometimes it is present, sometimes is missing and sometimes it is present but during the match it disappears.

Did you solve it? Does anyone have same problem?
thanks

**Comment by A1eNaz** [comment]
> Hi, i'm using the latest Canary version ( a6e3d77) and i have this issue: the game works fine but the soundtrack very often is missing. It is always missing in the opening screen (where it is necessary to push START), into the character select screen and into the game mode section. During the game, in some stages, sometimes, the soundtrack is missing: sometimes it is present, sometimes is missing and sometimes it is present but during the match it disappears.
> 
> Did you solve it? Does anyone have same problem? thanks

https://github.com/A1eNaz/Xenia-Game-Settings/wiki/Dead-or-Alive-4#recommended-settings-

**Comment by legolas119** [comment]
thank you very much! it works!
I activated the patch too, of the TU1 version. I tried to activate Alpha-152 unlocked , but when i select her into the characters select screen, the game freezes. Do you have this behaviour?
thanks

**Comment by A1eNaz** [comment]
> thank you very much! it works! I activated the patch too, of the TU1 version. I tried to activate Alpha-152 unlocked , but when i select her into the characters select screen, the game freezes. Do you have this behaviour? thanks

update your canary, it doesn't freeze on newer builds

**Comment by legolas119** [comment]
i downloaded the latest version, but unfortunately it still freezes:
![dead](https://github.com/user-attachments/assets/3785cbf6-8285-4ec2-8045-2425e96e7b36)

Do you know why?
thank you :)

**Comment by A1eNaz** [comment]
> i downloaded the latest version, but unfortunately it still freezes: ![dead](https://private-user-images.githubusercontent.com/48774386/399009242-3785cbf6-8285-4ec2-8045-2425e96e7b36.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzNzcyMDAsIm5iZiI6MTczNTM3NjkwMCwicGF0aCI6Ii80ODc3NDM4Ni8zOTkwMDkyNDItMzc4NWNiZjYtODI4NS00ZWMyLTgwNDUtMjQyNWU5NmU3YjM2LlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI4VDA5MDgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM5ZjFjYmQ5ZjRkYzc5MWU1Njc1N2RlNGEwYjMyZGQ4ODI2NjZlZmYyMGZlNjliNTc4NDdjYWE0NWQzOTBiYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Uy9ea1zqxfxdx43Z1V_Lg9Fqd-kgfW06aZ3cdQpYBMw)
> 
> Do you know why? thank you :)

try `protect_zero = false` and see if it works?

**Comment by legolas119** [comment]
yes, it works now! It isn't present  the portrait of the character but i think it is normal:
![Cattura](https://github.com/user-attachments/assets/49c1cd26-bec7-475f-bd62-55fd95566ba2)

thanks!

**Comment by A1eNaz** [comment]
> yes, it works now! It isn't present the portrait of the character but i think it is normal: ![Cattura](https://private-user-images.githubusercontent.com/48774386/399070483-49c1cd26-bec7-475f-bd62-55fd95566ba2.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzODI3NTYsIm5iZiI6MTczNTM4MjQ1NiwicGF0aCI6Ii80ODc3NDM4Ni8zOTkwNzA0ODMtNDljMWNkMjYtYmVjNy00NzVmLWJkNjItNTVmZDk1NTY2YmEyLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI4VDEwNDA1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwNDdmYTIxOTE5YWI4NTA1OTljYTkyODkwMDgyNjMwMTE0MDQ5Yjc0NTFkOTI1YmQwMTMwYmMxYTk5N2U5MzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fSXeuZqMTSrTiPBNpxUIpHLTZhCJ-IasD4yW5Dm0r3U)
> 
> thanks!

Press F5 or enable clear_memory_page_state

**Comment by legolas119** [comment]
yes, it is already enabled that option and i tried to push F5 during the game, but the portrait of alpha-152 is always missing...quite strange...do you see it?

**Comment by A1eNaz** [comment]
> yes, it is already enabled that option and i tried to push F5 during the game, but the portrait of alpha-152 is always missing...quite strange...do you see it?

its because the character wasnt supposed to be shown in the select screen ig, everyone else works fine tho

**Comment by legolas119** [comment]
ah ok ok! thanks!

**Comment by Xfitshadps44** [comment]
HOW TO REMOVE VERTEX EXPLOSIONS?

**Comment by A1eNaz** [comment]
> HOW TO REMOVE VERTEX EXPLOSIONS?

enable clear_memory_page_state
[Source](https://github.com/A1eNaz/Xenia-Game-Settings/wiki/Dead-or-Alive-4#recommended-settings-)

**Comment by Xfitshadps44** [comment]
> > HOW TO REMOVE VERTEX EXPLOSIONS?
> 
> set `clear_memory_page_state to true`. [Source](https://github.com/A1eNaz/Xenia-Game-Settings/wiki/Dead-or-Alive-4#recommended-settings-)


In general config settings?

**Comment by Xfitshadps44** [comment]
> > HOW TO REMOVE VERTEX EXPLOSIONS?
> 
> set `clear_memory_page_state to true`. [Source](https://github.com/A1eNaz/Xenia-Game-Settings/wiki/Dead-or-Alive-4#recommended-settings-)

![Image](https://github.com/user-attachments/assets/e90b9f51-ec6d-48f7-8af1-3a164df959da) Still glitches

**Comment by A1eNaz** [comment]
js join the discord server and get help there

**Comment by frannysunglasses** [comment]
I apologize if I'm being disrespectful or breaking any rules of the topic, but I would like to ask a question.

Is the game fully playable without any glitches or does it still have a lot of problems? I have a 3060 11800h laptop and I would like to know if it can handle emulating this game.

Thank you for your help and for your great commitment to making this game accessible.

**Comment by narojima** [comment]
> I apologize if I'm being disrespectful or breaking any rules of the topic, but I would like to ask a question.
> 
> Is the game fully playable without any glitches or does it still have a lot of problems? Do I just need to get the new/current version of Xenia Canary and do the manual configurations and it will work?
> 
> I have a 3060 11800h laptop and I would like to know if it can handle emulating this game.
> 
> Thank you for your help and for your great commitment to making this game accessible.

I feel like the game runs great, other then the graphical issues.

**Comment by arisupocket** [comment]
i am INCREDIBLY happy to announce that for the very first time, DOA4 runs on linux on xenia edge windows via proton.

<img width="2553" height="1597" alt="Image" src="https://github.com/user-attachments/assets/10890bbb-e931-4592-b40e-d1ffebe4c111" />\

i don't even know what to say beyond the fact that i am so beyond incredibly grateful. this is literally my favorite fighting game of all time and one of the few remaining reasons i keep a windows PC around... incredible work to the folks developing xenia edge.

**Comment by stepsy** [comment]
protect_zero = false
causes game to hang on second boot
current canary - e6828af

**Comment by DavidHale34** [comment]
Tested on latest release of Dec 15 for windows. Game runs perfectly on native resolution when "draw_resolution_scale" is set to 1. Smooth frame rate and high performance without graphic bugs (at least for the 2 hours tested), looks and feels almost identical to original. Seems like the game doesn't require non-default settings anymore, except "use_dedicated_xma_thread = false" to fix the background sound, which will be otherwise broken.

However, if we set "draw_resolution_scale" higher than 1 a few graphic bugs start to show. Specifically it's the water reflections and ripples on some stages ( Kyoto in Bloom and Experimental Playground). Also, the deforming sand on the Seaside Market will start to display incorrectly and there might be some occasional bugs that load corrupt/incorrect textures to overlay or fog effects. None of those appear at native resolution. Still, quite minor issues as the game looks and performs really well even on high resolution. Thanks for all the work.

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/dc3d7d4a-d899-45d8-9e23-330e879ce031" />

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/f6e15b6b-ab24-4e1f-9d78-c947c424290c" />
Most stages display almost perfectly at high-res.

![Image](https://github.com/user-attachments/assets/b616ff39-2025-49fc-b650-6f84385c32a4)
Broken reflections on some stages using high-res. Displays but wrong, gets broken on stage reload

**Comment by DavidHale34** [comment]
With the [latest ](https://github.com/xenia-canary/xenia-canary-releases/releases/tag/fcebdfb) update all of the high-res issues mentioned above have been fixed. Tested for 2 hours, no issues found. High performance, smooth framerate without bugs. The game requires only one non-default setting and two optional:

use_dedicated_xma_thread = false 
(to fix background sound)

draw_resolution_scale_x = 2
draw_resolution_scale_y = 2
(for higher render resolution)

I think it's fair to say now that the game achieved practically perfect emulation status. Big thanks to Margen67 and everyone else who made it possible.

![Image](https://github.com/user-attachments/assets/a82dc074-2e9b-4570-a3ca-9cf7b838e057)
The only visual bug I found which I think it's not related to emulation. Affects only 2 dinosaurs.

![Image](https://github.com/user-attachments/assets/5ed2fb78-0b22-4b07-ab95-d09cfed596ae)
Fully working reflections and ripples with high-res on all stages

![Image](https://github.com/user-attachments/assets/581057bf-d9bf-4981-8090-ac6847776045)

![Image](https://github.com/user-attachments/assets/10c77fe4-b3ea-4abb-9e1b-9b60405eacb8)
Correctly working post-process and screen space effects with high-res

**Comment by johnduharc** [comment]
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/9cdb1f03-457b-48da-941f-d50fc9d0d79b" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/1051797e-0a29-4398-8498-59f7ad4a81b5" /> is this still fixed? i'm getting these with the latest canary build, i am running this on an nvidia 3060.

**Comment by DavidHale34** [comment]
@johnduharc yes, it's fixed. Make sure you're running [this](https://github.com/xenia-canary/xenia-canary-releases/releases/tag/83d776f) build.

![Image](https://github.com/user-attachments/assets/38619074-d313-4957-8f94-20ed73525912)

**Comment by Moonalesca** [comment]
The game is almost perfect now, at long last. But I've encountered a game-breaking issue on the Temple on the Mountain stage. The characters can fall down the first set of steps just fine but it seems like the collision on the second steps is completely borked. Characters just fall through them and then tumble off into the distance forever. This is on Xenia Canary version 086decfa6.

**Comment by Moonalesca** [comment]
> The game is almost perfect now, at long last. But I've encountered a game-breaking issue on the Temple on the Mountain stage. The characters can fall down the first set of steps just fine but it seems like the collision on the second steps is completely borked. Characters just fall through them and then tumble off into the distance forever. This is on Xenia Canary version 086decfa6.

Update: I've moved over to my ROG Ally and this issue is not reproducible. My main desktop has a 21:9 monitor so I suspect that's the problem.

Update 2: Found the issue! One of the patches was causing it - specifically the “Increased Max Character Distance” patch. I turned that off and can now enjoy the game in ultrawide. Just be aware that the ultrawide patch throws Alpha-152’s outline off to the side which looks funky.

**Comment by Sowa95** [comment]
I'm offended by your laziness. All you have to do is run the game, same as you already do on Canary.

**Comment by RADIOACTIVEshrek** [comment]
> > Translation: I don't want to lose 15 minutes on testing a game I care about. I won't even attempt to run it myself, neither on Master nor Edge, because that would answer my own question. Instead I'll ask here and wait for days, or weeks for someone to give me the answer on a silver platter.
> 
> I've already played the game on Xenia Canary, bro. I just wanted to know how the game is performing on the standard Xenia version and the Edge version, which has gained a lot of traction recently in the emulation community.
> 
> I just want to know if it's worth migrating versions or sticking with Canary, that's all.
> 
> I don't understand why you're getting so offended by a genuine question.

Sorry about that asshole Sowa95 - I was offended by their smug rudeness and *their* laziness.
To answer your question 3 weeks late, Canary appears to run better from what I tested. If you run into any performance issues, just use DirectX 12 and turn off V-Sync, which also improved performance for me.

**Comment by RADIOACTIVEshrek** [comment]
> There's 40 messages of irrelevant rambling. This thread is anything but "informative". https://github.com/xenia-canary/game-compatibility?tab=readme-ov-file#what-not-to-post

I looked over the whole thread. This is a very normal thread, of which you're the primary outlier. I seriously hope you stop posting in any one of these threads because you seem to vastly prefer making whatever you see as a problem much worse rather than helping. I don't know how you can see your own behavior and not notice the issue.

To mitigate some of this, I'll add some info;
The Canary patch by ryu_hayabusa for making Alpha-152 playable has two invalid commands related to array address, but it doesn't seem to affect the character itself from my testing. This is in 02d2cb5cc.

Also, water is still broken, though notably for me it is broken no matter what is done (whether high-res with upscaling or original resolution).

**Comment by Moonalesca** [comment]
> > There's 40 messages of irrelevant rambling. This thread is anything but "informative". https://github.com/xenia-canary/game-compatibility?tab=readme-ov-file#what-not-to-post
> 
> I looked over the whole thread. This is a very normal thread, of which you're the primary outlier. I seriously hope you stop posting in any one of these threads because you seem to vastly prefer making whatever you see as a problem much worse rather than helping. I don't know how you can see your own behavior and not notice the issue.
> 
> To mitigate some of this, I'll add some info; The Canary patch by ryu_hayabusa for making Alpha-152 playable has two invalid commands related to array address, but it doesn't seem to affect the character itself from my testing. This is in 02d2cb5cc.
> 
> Also, water is still broken, though notably for me it is broken no matter what is done (whether high-res with upscaling or original resolution).

Can you post a screenshot of how the water is broken for you? It looks fine on my end, but it's been a while since I played the game on original hardware.
